src/LightStep/Tracer.cs

```diff
@@ -96,18 +97,39 @@ public async void Flush()
             if (_options.Run)
             {
                 // save current spans and clear the buffer
-                // TODO: add retry logic so as to not drop spans on unreachable satellite
-                _logger.Debug($"Flushing SpanData");
-                List<SpanData> currentSpans = new List<SpanData>();
+                ISpanRecorder currentBuffer;
                 lock (_lock)
                 {
-                    _logger.Debug($"Lock freed, getting current buffer.");
-                    currentSpans = _spanRecorder.GetSpanBuffer();
-                    _logger.Debug($"{currentSpans.Count} spans copied from buffer.");
+                    currentBuffer = _spanRecorder.GetSpanBuffer();
+                    _spanRecorder = new LightStepSpanRecorder();
```

Comment:
Since `_spanRecorder` always gets rewritten, it looks like the only record of `DroppedSpansCount` implicitly resets to `0` at the start of every `Flush`. Thus, if multiple requests fail in a row, the `DroppedSpanCount` sent on the next successful request will only reflect the last attempt. I believe we want to capture the full total.

(BTW, I might be totally mistaken about the behaviour, cuz C# ignoramus!)

Reply:
Ah, yeah, I should keep a running total because there's no guarantee that there won't be multiple in-flight attempts to send.